types/prop-types/index.d.ts

```diff
@@ -9,15 +9,27 @@ import { ReactNode, ReactElement } from 'react';
 
 export const nominalTypeHack: unique symbol;
 
-export type IsOptional<T> = undefined | null extends T ? true : undefined extends T ? true : null extends T ? true : false;
+export type IsOptional<T> = undefined | null extends T
```

Comment:
My editor auto-formatted these changes. I believe it's easier to read, but if need be I can revert.